### PR TITLE
feat(workflows): ground the create-time copilot — output-node delivery, teammate resolution, tool capability (#813)

### PR DESCRIPTION
**Pull request by oxoxDev**
## Summary

The copilot-grounding half of EPIC #813 (defects 1–3). Authoring end-to-end on staging took four runs and three edits to deliver one report — every failure was an authoring-surface problem, not an engine one. The create-time copilot is a one-shot LLM call steered by prose whose only enforcement folds a wrong draft to a silent `NotAutomatable`. Mirroring OpenHuman's reference host (host-side hard gates with model-consumable errors + rich grounding; **no** fuzzy resolver, **no** semantic tool matcher), with a hybrid **one-shot + at most one corrective re-prompt**.

- **Defect 1 — delivery expressed as an agent instruction, not an `output` node.** Only `output` nodes deliver; agents structurally can't send. `graph_contract` now states the invariant + shows an output-node example, and a **delivery gate** rejects a draft whose description asks to deliver (an address, a `#channel`, or "email/notify/DM me") but has no `output` node with a `destination`. The host never fabricates the destination — it tells the model to add one. Detector is conservative: a bare "we email customers weekly" produces no signal.
- **Defect 2 — a real-but-wrong teammate, and silent folds.** A deterministic **resolver** rewrites an `agent` id that isn't on the roster but uniquely matches one teammate's id/role/name (normalized), with an operator-facing note; no match or an ambiguous one becomes a gate error **naming the roster** — killing the old silent `NotAutomatable`. A "wrong-but-real" arm catches a draft that used a different real teammate than the one the request names.
- **Defect 3 — a legal tool that can't do the job, with empty args.** New `WORKFLOW_TOOL_CATALOG` gives each slug an honest capability line (`read_workspace_state` = "overview only … cannot read a file's contents") + `required_args`, presented to the model so it can pick a *capable* tool; `validate_tool_call_node` gains a **required-args-present** arm (args live under `config.args`).
- Correction **notes** flow to the console so the author sees why a draft was changed.

`Part of #813` — the first-run UX half (defects 5/7/8) is PR #814.

## API Or Behavior Changes

- **Author-time:** a `tool_call` node missing a required arg (per the catalogue) is now rejected at create/update (400) — for hand-authors too, since it lives in the shared gate. Saved graphs are not re-validated on load, so no load regression.
- **Copilot:** a rejected draft now triggers one corrective re-prompt (≤2 metered model calls under one run_id) and, on a second failure, a `NotAutomatable` naming the specific gate sentences instead of a bare fold. New `notes` field on the draft response.
- `WORKFLOW_TOOL_CATALOG` supersedes `WORKFLOW_TOOL_SLUGS` (internal; slug table kept as a `#[cfg(test)]` pin). No route/schema change.

## Tests

- [x] `cargo fmt --all -- --check` — pass
- [x] `cargo clippy --no-deps --features openhuman,tinycortex --all-targets -- -D warnings` — pass
- [x] `cargo check --all-targets` / `cargo check --all-features` — pass
- [x] Gated lib suite `--features openhuman,tinycortex` — 3533 passed / 0 failed (needs `RUST_MIN_STACK=16777216`; the default-stack `--tests` aborts on the unrelated pre-existing `harness::brain` stack-overflow). Independently re-verified the touched modules (workflow_build / workflow_create / caps): 176 passed / 0 failed.
- [x] Frontend: `npm run typecheck` · `npx vitest run` (580 passed, incl. new `workflow-draft.test.ts`) · `npm run build` — pass
- New Rust tests (gated lane) cover: normalize, conservative delivery signals, resolver rewrite+note, unknown→roster-naming error (silent-fold death), delivery gate pos/neg/silent, wrong-but-real, retry recovers, required-args missing/present; catalogue↔`namespace_of` coupling. Each gate observed RED against pre-change behaviour before restoring.

## Documentation

No doc file change — the grounding is code-internal and the new rules surface as author-facing error/notice text. Tool capability lines live in `WORKFLOW_TOOL_CATALOG`.

## Summary by CodeRabbit

- **New Features**
  - Workflow drafts can now include review summaries and correction notes from the drafting assistant.
  - Draft creation displays actionable notes when adjustments were made.
  - Draft generation can retry corrections automatically when workflow requirements are not met.

- **Bug Fixes**
  - Improved validation for required tool arguments, agent selection, delivery steps, and output configuration.
  - Enhanced workflow drafting guidance for available tools and roster information.

- **Tests**
  - Added coverage for draft notes, validation failures, corrective retries, and successful workflow delivery.

### CODE REVIEW

File: src/harness/workflow_build.rs [hunk 1]
```diff
@@ -1206,12 +1290,303 @@ pub(crate) enum DescriptionDraftOutcome {
...
+    for verb in ["email", "send", "notify", "dm", "message"] {
+        for object in ["me", "us"] {
+            let phrase = format!("{verb} {object}");
+            if lower.contains(&phrase) {
+                signals.push(format!("“{phrase}”"));
+            }
+        }
+    }
+    signals
+}
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply src/harness/workflow_build.rs:1509]: Addressed in 896c87a5. CHANNEL_RE now requires a leading letter so `#4521` no longer matches, and the verb+me/us check uses the whole-word `phrase_in` on normalized text so "send used parts" is not "send us". Added both cases to `delivery_signals_are_conservative`.

File: src/harness/workflow_build.rs [hunk 2]
```diff
@@ -1249,76 +1638,107 @@ pub(crate) async fn draft_workflow_from_description(
...
+    for _attempt in 1..=MAX_DRAFT_ATTEMPTS {
+        let (draft, usage) = match call_model(&builder, system.clone(), user).await {
+            Ok(pair) => pair,
+            Err(failure) => {
+                record_usage(runtime, &builder, COPILOT_AGENT, &run_id, &failure.usage).await;
+                // A model/parse failure is not a gate the model can correct — fold
+                // straight through, exactly as before #813.
+                return Ok(DescriptionDraftOutcome::NotAutomatable(failure.reason));
+            }
+        };
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply src/harness/workflow_build.rs:1679]: Addressed in 896c87a5. `call_model` now takes one shared `deadline`; the draft loop computes it once (`Instant::now()+BUILD_TIMEOUT`) and both attempts use `timeout_at`, so a retry shares the budget instead of adding a second full BUILD_TIMEOUT. The card pass passes `now+BUILD_TIMEOUT` (unchanged).
